synapse/util/caches/stream_change_cache.py

```diff
@@ -94,7 +95,17 @@ def get_entities_changed(
         """
         changed_entities = self.get_all_entities_changed(stream_pos)
         if changed_entities is not None:
-            result = set(changed_entities).intersection(entities)
+            # We now do an intersection, trying to do so in the most efficient
+            # way possible (some of these sets are *large*). First check in the
+            # given iterable is already set that we can reuse, otherwise we
+            # create a set of the *smallest* of the two iterables and call
+            # `intersection(..)` on it (this can be twice as fast as the reverse).
+            if isinstance(changed_entities, (set, frozenset)):
```

Comment:
Is this meant to be `entities` here? I think `changed_entities` will always be ` list` (`get_all_entities_changed` returns `None` or a `list`).

Reply:
whoops, yes